projecttacoma/fqm-execution
# Sort mustSupports, compare DRs with direct reference codes

# Summary
While working on data requirements tasking, Hoss and Lauren noticed some issues with the data requirements comparison script now that mustSupports are being incorporated. There were three issues and this addresses two of them: the mustSupport arrays were not being sorted before being compared (i.e. ['code', 'status'] did not equal ['status', 'code'] and data requirements that are not referenced by valueSets but rather through direct reference codes were not being analyzed for mustSupport comparison. 

## New behavior
- Sorts the mustSupports arrays before doing a comparison.
- Compares mustSupports of data requirements that are referenced by a direct reference code rather than a ValueSet. 

## Code changes
- `data-requirements/summary-compare.ts` - implemented above changes 

# Testing guidance
- `npm run check`
- Run the summary-compare script: `./summary-compare.sh`
- Make sure mustSupports pass when the arrays have the same contents 
- Make sure data requirements using direct reference codes are included in the must supports comparison
